inductiva/resources/machine_cluster.py

```diff
@@ -0,0 +1,86 @@
+"""Class to manage the MPI cluster in Google Cloud."""
+from absl import logging
+
+from inductiva.resources import machines_base
+
+
+class MPICluster(machines_base.BaseMachineGroup):
+    """Class to launch and manage an MPI cluster in Google Cloud.
+
+    A MPI cluster is a collection of homogenous machines all working together on
+    a common task given the configurations that are launched in Google Cloud.
+    Note: The cluster will be available only after calling 'start' method.
+    The billing will start only after the machines are started."""
+
+    def __init__(self,
+                 machine_type: str,
+                 num_machines: int,
+                 disk_size_gb: int = 70,
+                 register: bool = True) -> None:
+        """Create a MPICluster object.
+
+        The register argument is used to indicate if the machine group should
+        be registered or if it was already registered. If set as False on
+        initialization, then, the machine group is not registered and it
+        can not be started in the cloud. This serves has an helper argument for
+        retrieving already registered machine groups that can be started, for
+        example, when retrieving with the `machines_groups.get` method.
+        Users should not set this argument.
+
+        Args:
+            machine_type: The type of GC machine to launch. Ex: "e2-standard-4".
+              Check https://cloud.google.com/compute/docs/machine-resource for
+              information about machine types.
+            num_machines: The number of virtual machines to launch.
+            disk_size_gb: The size of the disk in GB, recommended min. is 60 GB.
+        """
+        super().__init__(machine_type=machine_type,
+                         disk_size_gb=disk_size_gb,
+                         register=register)
+        self.num_machines = num_machines
+        self.is_elastic = False
+        self.type = "mpi"
+
+        if register:
+            self._id, self._name = self._register_machine_group(
+                num_vms=self.num_machines,
+                is_elastic=self.is_elastic,
+                type=self.type)
+
+    @classmethod
+    def from_api_response(cls, resp: dict):
+        machine_group = super().from_api_response(resp)
+        machine_group.num_machines = int(resp["num_vms"])
+        machine_group.register = False
+        return machine_group
+
+    def start(self):
+        """Start the MPI Cluster."""
+        return super().start(num_vms=self.num_machines,
+                             is_elastic=self.is_elastic,
```

Comment:
It breaks because it won't have all the necessary parameters in the backend. Let's make it hidden for now and later we change all of this from the ground up (`BaseMachineGroup` I am looking at you).